__tests__/app.test.js

```diff
@@ -158,7 +157,59 @@ describe('GET/api/reviews/:review_id/comments', () => {
   })
 
 
-  
+
+describe.only('POST /api/reviews/:review_id/comments', () => {
+  test('responds with 201 and posted comment', () => {
+    return request(app)
+      .post('/api/reviews/3/comments')
+      .send({ username: 'bainesface', body: 'EPIC board game!!' })
+      .expect(201)
+      .then((res) => {
+        const { comment } = res.body;
+        console.log(comment);
+        expect(typeof comment).toBe('object');
+        expect(comment).toHaveLength(1);
+        comment.forEach((result) => {
+          expect(result).toEqual(
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String)
+            })
+          );
+        });
+      });
+    })
+    test('Responds with 404 if no ID match', () => {
+      return request(app)
+      .post('/api/reviews/999/comments')
+      .send({ username: 'bainesface', body: 'EPIC board game!!' })
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toEqual("Please provide a valid review_id");
+      })
+    })
+    test('Responds with 404 if no username match', () => {
+      return request(app)
+      .post('/api/reviews/3/comments')
+      .send({ username: 'Suuuuuiiiii', body: 'EPIC board game!!' })
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toEqual('Invalid Username');
+      });
+    });
+    test('Responds with 400 if invalid ID type', () => {
+      return request(app)
+      .post('/api/reviews/$$/comments')
+      .send({ username: 'bainesface', body: 'EPIC board game!!' })
+      .expect(400)
+      .then((res) => {
```

Comment:
please test for 201 ignores unnecessary properties, 400 missing properties, 404 non existent id